zkir/Utils/AssemblyFormatUtils.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2025 The ZKIR Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef ZKIR_UTILS_ASSEMBLYFORMATUTILS_H_
+#define ZKIR_UTILS_ASSEMBLYFORMATUTILS_H_
+
+#include "llvm/ADT/APInt.h"
+#include "llvm/ADT/bit.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/OpImplementation.h"
+#include "mlir/IR/Types.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir::zkir {
+
+template <typename T>
+Type parsePrimeField(AsmParser &parser) {
```

Comment:
How about `parseModulus`? It goes well with current syntax (e.g. `parseInteger`). Kind of hesitant to use `PrimeField` in generic context.

Reply:
Do you think it's good to have `parsePrimeField` in `ParserUtils.h`?